### PR TITLE
[FEATURE]: SdkManager implementation for services

### PR DESCRIPTION
**Pull request by PixoDev**
This PR provides a new system called SdkManager, this service holds factory logic to create and return SDK modules based on the credentials stored in memory.

With this we reduce singleton services in mobile and weird .initialize() methods or .instance static properties

### CODE REVIEW

File: src/services/photos/index.ts [hunk 1]
```diff
@@ -60,17 +62,13 @@ export class PhotosService {
...
+    const { results, count } = await this.sdk.photos.photos.getPhotos({}, skip, limit);
 
     return { results, count };
   }
 
   public async deletePhotos(photos: photos.Photo[]): Promise<void> {
-    this.checkModel();
-
-    Promise.all(photos.map(async (photo) => PhotosCommonServices.sdk.photos.deletePhotoById(photo.id)));
+    Promise.all(photos.map(async (photo) => this.sdk.photos.photos.deletePhotoById(photo.id)));
```
Comment (sg-gs) [inline src/services/photos/index.ts:71]: `return` or `await` this, if not, this promise is being fired without control

**Review by sg-gs — APPROVED** [review]

File: src/services/photos/index.ts (hunk 1 above)
Comment (PixoDev) [reply src/services/photos/index.ts]: Fixed, nice catch

File: src/services/photos/index.ts [hunk 2]
```diff
@@ -60,17 +62,13 @@ export class PhotosService {
...
+    const { results, count } = await this.sdk.photos.photos.getPhotos({}, skip, limit);
 
     return { results, count };
   }
 
   public async deletePhotos(photos: photos.Photo[]): Promise<void> {
-    this.checkModel();
-
-    Promise.all(photos.map(async (photo) => PhotosCommonServices.sdk.photos.deletePhotoById(photo.id)));
+    Promise.all(photos.map(async (photo) => await this.sdk.photos.photos.deletePhotoById(photo.id)));
```
Comment (sg-gs) [inline src/services/photos/index.ts:71]: `await` still missing

**Review by sg-gs — CHANGES_REQUESTED** [review]

File: src/services/photos/index.ts (hunk 2 above)
Comment (PixoDev) [reply src/services/photos/index.ts:71]: https://gyazo.com/ab14619ab1e46476edee257eaffdc434 ?

Comment (sg-gs) [reply src/services/photos/index.ts:71]: `await Promise.all(photos.map(async (photo) => await this.sdk.photos.photos.deletePhotoById(photo.id)));`

**Review by sg-gs — CHANGES_REQUESTED** [review]

**Review by sg-gs — APPROVED** [review]

**Pull request merged**